superset-frontend/src/explore/components/ControlPanelsContainer.tsx

```diff
@@ -276,21 +277,36 @@ export const ControlPanelsContainer = (props: ControlPanelsContainerProps) => {
   >(state => state.explore.controlsTransferred);
 
   useEffect(() => {
+    const removeDatasourceWarningFromControl = (
```

Comment:
nit: Could we define this outside in a `useCallback`?

Reply:
We could, but that wouldn't bring much value I think since we won't use it in any other place. I guess it's a style preference

Reply:
You're right, I've tended to err on the side of just spraying `useCallback` everywhere, but I agree it's not really needed in trivial cases like this where 1) creating it is dirt cheap 2) it's not being used in a prop.